theano/sandbox/cuda/corr_gemm.cu

```diff
@@ -54,6 +54,7 @@ inline int GET_BLOCKS(const int N) {
 // Kernels for fast unfold + copy
 __global__ void im2col_kernel(const int n, const float* data_im,
     const int height, const int width, const int kernel_h, const int kernel_w,
+    const int dilation_h, const int dilation_w,
```

Comment:
I've used Caffe code only in col2im_kernel in corr_gemm.cu  (the part of h_k ... ).
The rest of the code was written derived from the Theano code. But looking at Caffe code, there is high analogy between both codes. If you think we should copy exactly the code from Caffe then please confirm it to me and I shall copy it and re-submit.